tests/std/tests/P0040R3_parallel_memory_algorithms/test.cpp

```diff
@@ -0,0 +1,261 @@
+        auto buffer         = make_unique<unsigned char[]>(testSize);
+        const auto begin_it = reinterpret_cast<wrap_uchar*>(buffer.get());
+        const auto end_it   = begin_it + testSize;
+        uninitialized_value_construct(exec, begin_it, end_it);
+        assert(all_of(begin_it, end_it, expectation));
+    }
+};
+
+struct test_case_uninitialized_value_construct_n_parallel {
+    template <typename ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto buffer          = make_unique<unsigned char[]>(testSize);
+        const auto begin_it  = reinterpret_cast<wrap_uchar*>(buffer.get());
+        const auto end_it    = begin_it + testSize;
+        const auto result_it = uninitialized_value_construct_n(exec, begin_it, testSize);
+        assert(all_of(begin_it, end_it, expectation));
+        assert(end_it == result_it);
+    }
+};
+
+struct test_case_uninitialized_value_construct_memset_parallel {
+    template <typename ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto buffer         = unique_ptr<int>(new int[testSize]);
+        const auto begin_it = buffer.get();
+        const auto end_it   = begin_it + testSize;
+        uninitialized_value_construct(exec, begin_it, end_it);
+        assert(all_of(begin_it, end_it, expectation_zero));
+    }
+};
+
+struct test_case_uninitialized_value_construct_n_memset_parallel {
+    template <typename ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto buffer          = unique_ptr<int>(new int[testSize]);
+        const auto begin_it  = buffer.get();
+        const auto end_it    = begin_it + testSize;
+        const auto result_it = uninitialized_value_construct_n(exec, begin_it, testSize);
+        assert(all_of(begin_it, end_it, expectation_zero));
+        assert(end_it == result_it);
+    }
+};
+
+struct test_case_destroy_parallel {
+    template <typename ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto buffer         = make_unique<unsigned char[]>(testSize);
+        const auto begin_it = reinterpret_cast<wrap_uchar*>(buffer.get());
+        const auto end_it   = begin_it + testSize;
+
+        uninitialized_default_construct(exec, begin_it, end_it);
+        destroy(exec, begin_it, end_it);
+    }
+};
+
+struct test_case_destroy_n_parallel {
+    template <typename ExecutionPolicy>
+    void operator()(const size_t testSize, const ExecutionPolicy& exec) {
+        auto buffer         = make_unique<unsigned char[]>(testSize);
+        const auto begin_it = reinterpret_cast<wrap_uchar*>(buffer.get());
+        const auto end_it   = begin_it + testSize;
+
+        uninitialized_default_construct_n(begin_it, testSize);
+        const auto result_it = destroy_n(exec, begin_it, testSize);
+        assert(end_it == result_it);
+    }
+};

```

Comment:
I think we're missing coverage for the non-trivially-destructible cases in `destroy` and `destroy_n`.